HeterogeneousCore/CUDAUtilities/src/allocate_device.cc

```diff
@@ -0,0 +1,40 @@
+#include <limits>
+
+#include "FWCore/Utilities/interface/Likely.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/ScopedSetDevice.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/allocate_device.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+#include "getCachingDeviceAllocator.h"
+
+namespace {
+  const size_t maxAllocationSize =
+      notcub::CachingDeviceAllocator::IntPow(cudautils::allocator::binGrowth, cudautils::allocator::maxBin);
+}
+
+namespace cudautils {
+  void *allocate_device(int dev, size_t nbytes, cudaStream_t stream) {
+    void *ptr = nullptr;
+    if constexpr (cudautils::allocator::useCaching) {
```

Comment:
In this way, not really.

Reply:
We used to have a runtime flag for this, when the allocators were part of the `CUDAService`.
If we ever want to re-introduce the possibility to disable the caching at runtime, rather than compile time, we could add some hooks there.